cdk_chalice/chalice.py

```diff
@@ -118,10 +118,12 @@ def _update_sam_template(self, sam_package_dir):
 
         with open(sam_template_path) as sam_template_file:
             sam_template = json.load(sam_template_file)
-            properties = sam_template['Resources']['APIHandler']['Properties']
-            properties['CodeUri'] = {
-                'Bucket': sam_deployment_asset.s3_bucket_name,
-                'Key': sam_deployment_asset.s3_object_key
-            }
+            functions = [v for k, v in sam_template['Resources'].items() if v['Type'] == 'AWS::Serverless::Function']
+            for function in functions:
+                properties = function['Properties']
+                properties['CodeUri'] = {
+                    'Bucket': sam_deployment_asset.s3_bucket_name,
+                    'Key': sam_deployment_asset.s3_object_key
+                }
```

Comment:
May I suggest a bit more verbose implementation to avoid single-letter variable names + keep the code within 84 column length for vertical readability (I still didn't document it, but this is the project standard - will add pep8 test for it later down the road)?

```python
            functions = filter(
                lambda resource: resource['Type'] == 'AWS::Serverless::Function',
                sam_template['Resources'].values()
            )
            for function in functions:
                function['Properties']['CodeUri'] = {
                    'Bucket': sam_deployment_asset.s3_bucket_name,
                    'Key': sam_deployment_asset.s3_object_key
                }
```